pkg/querier/http.go

```diff
@@ -87,8 +87,37 @@ func (q *QueryHandlers) Render(w http.ResponseWriter, req *http.Request) {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		return
 	}
+
+	// TODO: where to do this?
+	timelineStep := 10
+	series, err := q.upstream.SelectSeries(req.Context(),
+		connect.NewRequest(&querierv1.SelectSeriesRequest{
+			ProfileTypeID: selectParams.ProfileTypeID,
+			LabelSelector: selectParams.LabelSelector,
+			Start:         selectParams.Start,
+			End:           selectParams.End,
+			Step:          float64(timelineStep),
+		}))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
```

Comment:
We should fire those in parallel maybe ?

Reply:
using `g, ctx := errgroup.WithContext(ctx)` and then Go() for each upstream call.